### PR TITLE
cockpit: fix reprocess transactions payload

### PR DESCRIPTION
**Pull request by 0liveira**
## Contexto

Após o redirecionamento dos usuários da dashboard para a pilot, notamos que algumas transações falhavam durante reprocessar. Investigando, chegamos a conclusão que isso só ocorria ao reprocessar transações com o antifraude ativo e originárias da api versão `2013-03-01`.

Isso ocorre devido a diferença na estrutura de requisição feita pela api `2013-03-01`, que exige que as informações do `customer` estejam preenchidas (não aceita apenas o customer.id).
## Checklist
- [x] Adicionar suporte a versão 2013-03-01 da api

## Issues linkadas
- [x] Resolves https://github.com/pagarme/pilot/issues/1427

## Como testar?

Reprocessar uma transação com antifraude ativo e com a versão da api 2013-03-01. O comportamento esperado é que a transação seja reprocessada com sucesso.

### CODE REVIEW

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js [hunk 1]
```diff
@@ -33,15 +33,30 @@ const getCardId = ifElse(
...
 const getCustomerId = pathOr(null, ['customer', 'id'])
 
-const getCustomer = ifElse(
-  propEq('customer', null),
-  always(null),
-  pipe(
-    prop('customer'),
-    pick(['id'])
-  )
+const hasCustomerDocuments = hasPath(['customer', 'documents', 0])
```
Comment (miglsoares) [inline packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: por que desse `0`?

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js [hunk 2]
```diff
@@ -33,15 +33,30 @@ const getCardId = ifElse(
...
+
+const customerBuilder = ifElse(
+  hasCustomerDocuments,
+  applySpec({
+    id: path(['customer', 'id'])
+  }),
+  buildCustomerObject
 )
 
+const getCustomer = (transaction) => {
```
Comment (vagnervst) [inline packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: Pode usar hype nessa função também, caso queira :eyes:

```js
const getCustomer = ifElse(
  pipe(
    prop('customer'),
    isNil,
  ),
  always(null),
  customerBuilder
)
```

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js [hunk 3]
```diff
@@ -33,15 +33,30 @@ const getCardId = ifElse(
...
+  if (transaction.customer === null) {
+    return null
+  }
+
+  return customerBuilder(transaction)
+}
```
Comment (Deivis) [inline packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: Dá para tocar um ramdinha aqui, ex:
```suggestion
const hasCustomer = pipe(
  prop('customer'),
  complement(isNil)
)

const g = ifElse(
  hasCustomer,
  customerBuilder,
  always(null)
)
```

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js (hunk 1 above)
Comment (0liveira) [reply packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: Porque esse documents aí é um array e sem o `0` só estaríamos verificando se a propriedade existe.

Comment (miglsoares) [reply packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: ah sim, valeu!

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js (hunk 3 above)
Comment (0liveira) [reply packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: Fino, gostei.

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.js (hunk 2 above)
Comment (0liveira) [reply packages/cockpit/src/transactions/reprocess/cleanTransaction.js]: Boa!

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8eef576396808d8495240918b8faaf9b22acb61f
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8eef576396808d8495240918b8faaf9b22acb61f
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8eef576396808d8495240918b8faaf9b22acb61f
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8eef576396808d8495240918b8faaf9b22acb61f
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: packages/cockpit/src/transactions/reprocess/cleanTransaction.test.js [hunk 4]
```diff
@@ -21,4 +29,21 @@ describe('Reprocess', () => {
...
+
+    const toRequestExpected = pipe(
+      assoc('customer', customerObject),
+      assocPath(['customer', 'documents'], []),
+      dissocPath(['customer', 'born_at']),
+      dissocPath(['customer', 'document_number']),
+      dissocPath(['customer', 'gender']),
+      dissocPath(['customer', 'phone', 'id']),
+      dissocPath(['customer', 'phone', 'object']),
+    )(toRequest)
```
Comment (Deivis) [inline packages/cockpit/src/transactions/reprocess/cleanTransaction.test.js]: Ó as ideias dos ramda gostoso:
```js
const createCustomerObject = ({ address, customer, phone }) => ({
  address,
  phone,
  ...customer,
})

const removePhoneUnusedProps = pipe(
  dissocPath(['phone', 'id']),
  dissocPath(['phone', 'object'])
)

const removeUnusedCustomerProps = pipe(
  omit([
    'born_at',
    'document_number',
    'gender',
  ]),
  removePhoneUnusedProps
)

const addDcoumentsProp = assoc('documents', [])

const buildCustomerObject = pipe(
  createCustomerObject,
  removeUnusedCustomerProps,
  addDcoumentsProp
)

...

const customerObject = buildCustomerObject(fromRequest)

const toRequestExpected = assoc('customer', customerObject, toRequest)

```

Comment (0liveira) [reply packages/cockpit/src/transactions/reprocess/cleanTransaction.test.js]: Muito que gostei, @Deivis :tada:

**Pull request merged**